packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -137,9 +137,9 @@ class ThemeData {
     accentIconTheme ??= accentIsDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
     platform ??= defaultTargetPlatform;
     final Typography typography = new Typography(platform: platform);
-    textTheme ??= isDark ? typography.white : typography.black;
-    primaryTextTheme ??= primaryIsDark ? typography.white : typography.black;
-    accentTextTheme ??= accentIsDark ? typography.white : typography.black;
+    textTheme ??= (isDark ? typography.white : typography.black).apply(decoration: TextDecoration.none);
+    primaryTextTheme ??= (primaryIsDark ? typography.white : typography.black).apply(decoration: TextDecoration.none);
+    accentTextTheme ??= (accentIsDark ? typography.white : typography.black).apply(decoration: TextDecoration.none);
```

Comment:
how would you feel about either removing these additions (relying on the way we inherit from a base that doesn't have an underline) or hard-coding them into the constants? It seems weird to have these done in the middle-ground manner.